server/storage/wal/wal.go

```diff
@@ -962,6 +962,11 @@ func (w *WAL) Save(st raftpb.HardState, ents []raftpb.Entry) error {
 	}
 
 	mustSync := raft.MustSync(st, w.state, len(ents))
+	// We should sync the HardState to disk as long as it differs from previous value.
+	// See discussion in https://github.com/etcd-io/raft/pull/300.
+	if st != w.state {
```

Comment:
I'm worried about overhead of forcing sync for every entry just to fix `--force-new-cluster`. From what I understand this will introduce syncing every time commit index increases.

Have you tried suggestion by @pav-kv https://github.com/etcd-io/raft/pull/300#issuecomment-2910081042?

Reply:
Updated